app/domain/authentication/authn_gcp/validate_resource_restrictions_configuration.rb

```diff
@@ -0,0 +1,49 @@
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of validating resource restrictions combination.
+    # the allowed combination is: Role should have at least 1 permitted constraints, and non of illegal constraints
+    ValidateResourceRestrictionsConfiguration = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       %i(resource_restrictions permitted_constraints)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ValidatingResourceRestrictionsConfiguration.new)
+        validate_restrictions_exist
+        validate_constraints_are_permitted
+        validate_restrictions_values_exist
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ValidatedResourceRestrictionsConfiguration.new)
+      end
+
+      private
+
+      def validate_restrictions_exist
+        if @resource_restrictions.nil? || @resource_restrictions.empty?
+          raise Errors::Authentication::RoleMissingRequiredConstraints.new(@permitted_constraints)
+        end
+      end
+
+      def validate_constraints_are_permitted
+        @resource_restrictions.each do |r|
+          resource_type = r.type
+          unless @permitted_constraints.include?(resource_type)
+            raise Errors::Authentication::ConstraintNotSupported.new(resource_type, @permitted_constraints)
+          end 
+        end
+      end
+
+      def validate_restrictions_values_exist
+        @resource_restrictions.each do |r|
+          resource_type = r.type
+          resource_value = r.value
+          if resource_value.nil? || resource_value.empty?
```

Comment:
when would this happen? can you share a host policy definition that would yield this error?

Reply:
Yes, the host policy will be like:
```
  - !host
    id: hosttest1
    annotations:
      description: nessi temp host
      authn-gcp/instance-name: 
```
The logs will looks like:
```
CONJ00039D Extracting resource restrictions from role 'host/nessitest/hosttest1' with prefix 'authn-gcp/'
  Sequel::Postgres::Database (0.4ms)  SELECT * FROM "resources" WHERE "resource_id" = 'cucumber:host:nessitest/hosttest1'
  Sequel::Postgres::Database (0.5ms)  SELECT * FROM "annotations" WHERE ("annotations"."resource_id" = 'cucumber:host:nessitest/hosttest1')
CONJ00024D Retrieved value of annotation authn-gcp/instance-name
CONJ00040D 1 resource restrictions were extracted
CONJ00041D Validating resource restrictions configuration
  Sequel::Postgres::Database (0.4ms)  SELECT * FROM "roles" WHERE ("role_id" = 'cucumber:host:nessitest/hosttest1') LIMIT 1
Authentication Error: #<Errors::Authentication::MissingResourceRestrictionsValue: CONJ000068E Resource restriction 'authn-gcp/instance-name' value  is missing or empty>
```

Reply:
thanks for clarifying